tests/templates/test_subroutines.py

```diff
@@ -369,21 +369,22 @@ def test_single_ex_unitary_operations(self, ph, ref_gates):
         with qml.utils.OperationRecorder() as rec:
             SingleExcitationUnitary(weight, wires=ph)
 
-        idx = ref_gates[0][0]
+        assert len(rec.queue) == sqg + cnots            
 
-        exp_gate = ref_gates[0][1]
-        res_gate = rec.queue[idx]
+        for gate in ref_gates:
+            idx = gate[0]
 
-        exp_wires = ref_gates[0][2]
-        res_wires = rec.queue[idx]._wires
+            exp_gate = gate[1]
+            res_gate = rec.queue[idx]
+            assert isinstance(res_gate, exp_gate)
 
-        exp_weight = ref_gates[0][3]
-        res_weight = rec.queue[idx].parameters        
+            exp_wires = gate[2]
```

Comment:
Same point here: Is it necessary to define `exp_wires`? Maybe just use `gate[2]` in the `assert`?

Reply:
I took this style from other unit tests and I think it is convenient. It helps to understand the assertion statement involving the "expected wires" ( ``exp_wires``) and "resulting wires" (``res_wires``). It also clarifies who is who in the ``ref_gates``.